spec/services/imo_api_service_spec.rb

```diff
@@ -85,6 +98,29 @@
           service.send_invitation(patient)
         }.to raise_error(ImoApiService::Error)
       end
+
+      it "overrides the patient's phone number with an optional phone number argument" do
+        phone_number = "+9991112223333"
+        locale = "bn-BD"
+        request_body = JSON(
+          phone: phone_number,
+          msg: I18n.t("notifications.imo.invitations.message", patient_name: patient.full_name, locale: locale),
+          contents: [{
+            key: I18n.t("notifications.imo.invitations.message_key", locale: locale),
+            value: I18n.t("notifications.imo.invitations.message", patient_name: patient.full_name, locale: locale)
+          }],
+          title: I18n.t("notifications.imo.invitations.title", locale: locale),
+          action: I18n.t("notifications.imo.invitations.action", locale: locale),
+          callback_url: "https://localhost/api/v3/patients/#{patient.id}/imo_authorization"
+        )
+
+        stub_request(:post, request_url).with(headers: request_headers, body: request_body).to_return(status: 200, body: success_body)
```

Comment:
I'm trying to remember the Webmock details, but I am pretty sure this will _not_ verify the phone number is passed through in the body for the request made.  It just sets a stub, not a verified mock.

It looks like you can pretty easily verify with https://github.com/bblimke/webmock#setting-expectations-in-rspec-on-the-stub.

So grab the stub, and verify after the API is called.

Reply:
That's an excellent point. These tests could easily continue passing without verifying the phone number portion if we removed the underlying web request. Thanks for pointing me in a better direction. Fixed it.